src/decoder/symphonia.rs

```diff
@@ -97,7 +104,10 @@ impl SymphoniaDecoder {
         let decoded = loop {
             let current_frame = match probed.format.next_packet() {
                 Ok(packet) => packet,
-                Err(_) => break decoder.last_decoded() // IoError end of stream is expected
+                Err(e) => match e {
```

Comment:
You can inline this match so:
Err(Error::IoError(_)) => break etc etc,
Err(other) => return Err(other),